niceman/resource/session.py

```diff
@@ -375,19 +384,46 @@ def read(self, path):
     def mkdir(self, path, parents=False):
         """Create a directory
         """
-        self.execute_command(["mkdir"] + (["-p"] if parents else [""]) + [path])
+        command = ["mkdir"]
+        if parents: command.append("-p")
+        command += [path]
+        self.execute_command(command)
 
         if not self.isdir(path):
             raise CommandError(cmd='mkdir', msg="Failed to create directory")
 
     def isdir(self, path):
         try:
-            self.execute_command(['test', '-d', path])
+            out, err = self.execute_command(['test', '-d', path])
+        except Exception as exc:  # TODO: More specific exception?
+            lgr.debug("Check for directory failed: %s", exc_str(exc))
+            return False
+        if not err:
             return True
-        except CommandError:
+        else:
+            lgr.debug("Standard error was not empty (%r), thus assuming that "
+                      "test for directory has failed", err)
             return False
-    # chmod?
-    # chown?
+
+    def chmod(self, mode, remote_path, recursive=False):
+        """Set the mode of a remote path
+        """
+        command = ['chmod']
+        if recursive: command += ["-R"]
+        command += [mode] + [remote_path]
+        self.execute_command(command)
+
+    def chown(self, uid, gid, remote_path, recursive=False):
```

Comment:
What is the best way to install @borrowdoc

I am poking around http://www.pymvpa.org/index.html

Reply:
`from niceman.dochelpers import borrowdoc`  or by "install" did you mean "use"? ;)

Reply:
as for use, its docstring is even helpful ;)
```ipython
In [1]: from niceman.dochelpers import borrowdoc

In [2]: borrowdoc?
Signature: borrowdoc(cls, methodname=None)
Docstring:
Return a decorator to borrow docstring from another `cls`.`methodname`

It should not be used for __init__ methods of classes derived from
ClassWithCollections since __doc__'s of those are handled by the
AttributeCollector anyways.

Common use is to borrow a docstring from the class's method for an
adapter function (e.g. sphere_searchlight borrows from Searchlight)

Examples
--------
To borrow `__repr__` docstring from parent class `Mapper`, do::

   @borrowdoc(Mapper)
   def __repr__(self):
       ...
```
so in many cases you just need to decorate the method stating which class to borrow from.

Reply:
I meant "install", not "use". Looks like we already have it from the datalad clone. :)
Am working on beefing up documentation and am loathe to copy/paste.